include/numerics/sparse_matrix.h

```diff
@@ -171,6 +171,13 @@ class SparseMatrix : public ReferenceCountedObject<SparseMatrix<T> >,
   virtual void close () = 0;
 
   /**
+   *  For PETSc matrix , this function is similar to close but without shrinking memory.
+   *  This is useful when we want to switch between ADD_VALUES and INSERT_VALUES.
+   *  close should be called before using the matrix.
+   */
+  void flush () { close(); }
```

Comment:
This needs to be marked `virtual`

Reply:
Huh.  How did the libmesh_override in the subclass not catch that?

Reply:
Oh, it *did* catch that, just not in the No-C++11 build.